pandas/util/testing.py

```diff
@@ -867,18 +877,20 @@ def assert_interval_array_equal(left, right, exact="equiv", obj="IntervalArray")
         Specify object name being compared, internally used to show appropriate
         assertion message
     """
-    _check_isinstance(left, right, IntervalArray)
-
     assert_index_equal(
         left.left, right.left, exact=exact, obj="{obj}.left".format(obj=obj)
     )
     assert_index_equal(
         left.right, right.right, exact=exact, obj="{obj}.left".format(obj=obj)
     )
+    left = cast(IntervalArray, left)
```

Comment:
Are these casts required? Seems like they only subsequently get sent to `assert_attr_equal` which is untyped, so not sure what the failure would be (may also be overlooking)

Reply:
these casts are not required here.

```
$ mypy pandas --warn-redundant-casts
pandas\util\testing.py:886: error: Redundant cast to "IntervalArray"
pandas\util\testing.py:887: error: Redundant cast to "IntervalArray"
```